### PR TITLE
add test for complex relationships

### PR DESCRIPTION
**Pull request by zenchild**
## Priority
_Is this PR blocking your next action?_

No but the fix in #154 is.

## What Changed & Why

I had originally added a fix for infinite looping that was different than the one in #154 but I liked that approach better so I removed my code, merged in the fix from #154 but my test was still relevant so I added that back in.

## Testing
Assuming you have my branch pulled down you can see the infinite loop bug in action by doing the following:

```
# Check out the files prior to the fix by @Kokogino
git checkout 015a3799~1 -- src/middleware/json-api/_deserialize.js src/middleware/json-api/res-deserialize.js
./node_modules/.bin/mocha -g 'should deserialize complex relations without going into an infinite loop'
# When finished
git checkout HEAD src/
./node_modules/.bin/mocha -g 'should deserialize complex relations without going into an infinite loop'
```

## Bug/Ticket Tracker
Original Fix: #154 

## People
@Kokogino fixed the bug in #154

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "devour-client",
-  "version": "2.0.13",
+  "name": "@thebigknow/devour-client",
+  "version": "2.0.14",
```
Comment (tomers) [inline package.json]: Please remove all these changes in `package.json`

**Comment by Auspicus** [comment]
@zenchild This is just to add tests right? Do you mind re-rolling the latest master into this?

**Comment by zenchild** [comment]
@Auspicus  Sorry for the tardiness. I went on vacation right after you sent your message. :)

**Comment by Auspicus** [comment]
@zenchild No worries. Thanks for the contribution 🎉

**Comment by Auspicus** [comment]
Released in 2.0.27 🚀

**Pull request merged**